### PR TITLE
Add Groovy unit tests support

### PR DESCRIPTION
**Pull request by laurglia**

### CODE REVIEW

**Comment by laurglia** [comment]
I had two questions which maybe reviewers know the answer to:
1. How is pipeline-lib currently compiled? I didn't find anything in this repository about it.
2. What versions of dependencies do we actually use? There probably won't be much difference since I'm only adding tests for things that do not use these dependencies but perhaps would still be good to use the same version.

**Comment by indrekj** [comment]
> I had two questions ...

I have no idea.

**Comment by kuzaxak** [comment]
> How is pipeline-lib currently compiled? I didn't find anything in this repository about it.

It isn't compiled in any way. The shared library is a runtime library that is used in every Jenkinsfile across Glia. 

> What versions of dependencies (snakeyaml, pipeline-github, ...) do we actually use?

That information can be found in Jenkins plugins versions.

**Comment by laurglia** [comment]
Replaced @indrekj with @s22su because Indrek is offline and he did not post his status update in #tm-inf thread, so assuming that he is out today.

**Review by s22su — APPROVED** [review]
Seems legit and the tests run during the build.

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      srcDirs = ['src/']
+    }
+  }
+
+  test {
+    groovy {
+      srcDirs = ['test/']
+    }
+  }
+}
```
Comment (gugalo) [inline build.gradle:39]: NIT: If I understood correctly then I think that these lines should be removed and instead the project structure should be fixed.
`src/`  => `src/main/groovy`
`test/` => `src/test/groovy`

**Review by gugalo — APPROVED** [review]
If it works then it works. However, I added my 5 ceents.

File: build.gradle (hunk 1 above)
Comment (laurglia) [reply build.gradle:39]: This project is loaded to Jenkins as a shared library. In the [documentation](https://www.jenkins.io/doc/book/pipeline/shared-libraries/#directory-structure), they actually say to use this directory structure, so I think we can stick to that.

Comment (gugalo) [reply build.gradle:39]: Makes sense 👍

**Comment by laurglia** [comment]
!merge

**Pull request merged**